modules/openapi-generator/src/main/resources/python/api.mustache

```diff
@@ -35,307 +40,237 @@ class {{classname}}:
         self.api_client = api_client
 {{#operation}}
 
+
     @validate_call
-    {{#asyncio}}async {{/asyncio}}def {{operationId}}(
-        self,
-        {{#allParams}}
-        {{paramName}}: {{{vendorExtensions.x-py-typing}}}{{^required}} = None{{/required}},
-        {{/allParams}}
-        **kwargs,
-    ) -> {{{returnType}}}{{^returnType}}None{{/returnType}}:
-        """{{#isDeprecated}}(Deprecated) {{/isDeprecated}}{{{summary}}}{{^summary}}{{operationId}}{{/summary}}  # noqa: E501
+    {{#asyncio}}async {{/asyncio}}def {{operationId}}{{>partial_api_args}} -> {{{returnType}}}{{^returnType}}None{{/returnType}}:
+{{>partial_api}}
+        response_data = {{#asyncio}}await {{/asyncio}}self.api_client.call_api(
+            *_param,
+            _request_timeout=_request_timeout
+        )
+        {{#asyncio}}await {{/asyncio}}response_data.read()
+        return self.api_client.response_deserialize(
+            response_data=response_data,
+            response_types_map=_response_types_map,
+        ).data
```

Comment:
Could the bare `get` method return `get_with_http_info(...).data` to avoid duplicating this code?

Reply:
As you probably know, validate_arguments cannot be called twice. #16776
`raw_function` is of type Any.

> I'll try to think about something else, if we split the body of the functions as i proposed in #16788 , maybe that can offer an alternative that could also be useful to you, while keeping mypy happy :)

I think I have implemented what is being proposed here.

Reply:
Ahh right, thanks for clarifying! I knew I had seen this proposed somewhere before!

Here's another idea: What do you think of using two variables for the validated and unvalidated function, one being private? (also tagging @multani)

```py
def _get_with_http_info(...):
    ...

get_with_http_info = validate_call(_get_with_http_info)

def get(...):
    return _get_with_http_info(...).data
```

It's somewhat cosmetic, because it will only reduce the amount of code, but not change any external interface. Feel free to stay with the current solution.

Reply:
I think I had something more like what @robertschweizer is proposing in mind, but once the typing is correct, it wouldnalso be easier to refactor the internal implementation if needed.

Also, I think I would write the later proposal if I was writing the code manually as it has less duplication. Since the code is autogenerated, maybe it doesn't matter too much. Let's keep the current version :+1: